Explorer/Assets/Scripts/CrdtEcsBridge/ComponentWriter/ECSToCRDTWriter.cs

```diff
@@ -0,0 +1,76 @@
+using CRDT;
+using CRDT.Memory;
+using CRDT.Protocol;
+using CRDT.Protocol.Factory;
+using CrdtEcsBridge.Components;
+using CrdtEcsBridge.OutgoingMessages;
+using Google.Protobuf;
+using System;
+using System.Buffers;
+
+namespace CrdtEcsBridge.ECSToCRDTWriter
+{
+    public class ECSToCRDTWriter : IECSToCRDTWriter
+    {
+        private readonly ICRDTProtocol crdtProtocol;
+        private readonly IOutgoingCRTDMessagesProvider outgoingCRDTMessageProvider;
+        private readonly ISDKComponentsRegistry componentsRegistry;
+        private readonly ICRDTMemoryAllocator pooledMemoryAllocator;
+
+        public ECSToCRDTWriter(ICRDTProtocol crdtProtocol, IOutgoingCRTDMessagesProvider outgoingCRDTMessageProvider,
+            ISDKComponentsRegistry componentsRegistry)
+        {
+            this.crdtProtocol = crdtProtocol;
+            this.outgoingCRDTMessageProvider = outgoingCRDTMessageProvider;
+            this.componentsRegistry = componentsRegistry;
+            pooledMemoryAllocator = new CRDTPooledMemoryAllocator();
+        }
+
+        public void PutMessage<T>(CRDTEntity crdtID, int componentId, T model) where T: IMessage<T>
+        {
+            if (componentsRegistry.TryGet(componentId, out SDKComponentBridge componentBridge))
+            {
+                IMemoryOwner<byte> memory = pooledMemoryAllocator.GetMemoryBuffer(model.CalculateSize());
+                componentBridge.Serializer.SerializeInto(model, pooledMemoryAllocator.GetMemoryBuffer(model.CalculateSize()).Memory.Span);
+                ProcessMessage(crdtProtocol.CreatePutMessage(crdtID, componentId, memory));
+            }
+            else
+                throw new Exception($"Serializer not present for type {typeof(T).Name}");
```

Comment:
I think it was a good indicator in case something was not behaving as expected, it will give you a starting point on what may be going on. Maybe its better to change it to a Log?